ecpp

```diff
@@ -112,7 +140,102 @@ def main():
     except KeyboardInterrupt:
         print(n)
         raise
-    print("Number is prime")
+    print("\n")
+    print("[+] number is prime\n")
+
+def correlate(moduli, certs=None, verify_opt=None, primo_opt=None):    
+    print(primo_opt)
+    cert_files = []
+    if certs is None:
+        certs = certs_dir
+    for path, subdirs, files in os.walk(certs):
+        for name in files:
+            cert_files.append(os.path.join(path, name))
+    cert_for_moduli  = []
+    no_cert_for_moduli = []
+    status = None
+    with open(str(moduli)) as f:
+        inputfile = filter(None, (line.rstrip() for line in f))
+        count_line = 1
+        for line in inputfile:
+            if not line.startswith('#'):
+                prime_moduli = (line.split(" ")[6])
+                int_prime_moduli = int(prime_moduli, 16)
+                status = False
+                print("=====================================================")
+                print("Searching cert for: " + str(int_prime_moduli))
+                print("Moduli prime on line no: " + str(count_line))
+                for cert in cert_files:
```

Comment:
this has quadratic complexity

please load primality certs first and then compare the in-memory list with lines from moduli